core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2915,6 +2944,52 @@ wasi_ssp_sock_bind(
     return __WASI_ESUCCESS;
 }
 
+__wasi_errno_t
+wasi_ssp_addr_resolve(
+#if !defined(WASMTIME_SSP_STATIC_CURFDS)
+    struct fd_table *curfds,
+#endif
+    const char *host, const char *service, __wasi_addr_info_hints_t *hints,
+    __wasi_addr_info_t *addr_info, __wasi_size_t addr_info_size,
+    __wasi_size_t *max_info_size)
+{
+    wamr_addr_info_t *wamr_addr_info =
+        (wamr_addr_info_t *)malloc(addr_info_size * sizeof(wamr_addr_info_t));
```

Comment:
Don't use malloc, use wasm_runtime_malloc/wasm_runtime_free instead, some platforms may not support malloc/free.